app/src/main/java/org/stepik/android/presentation/step/StepPresenter.kt

```diff
@@ -154,17 +158,14 @@ constructor(
             .doOnSubscribe { isBlockingLoading = true }
             .doFinally { isBlockingLoading = false }
             .subscribeBy(
-                onSuccess = {
-                    val action = if (state.lessonData.isDemo && !it.isDemo) {
-                        navigationActionMapper.mapToCoursePurchaseAction(state.lessonData.course)
-                    } else {
-                        navigationActionMapper.mapToShowLessonAction(
-                            stepNavigationDirection,
-                            lessonData = it,
-                            isAutoplayEnabled = isAutoplayEnabled
-                        )
-                    }
-                    view?.handleNavigationAction(action)
+                onSuccess = { targetLessonData ->
+                    val requiredSectionSource =
```

Comment:
`getLessonDataForDirection` should return all required info to map it to navigation action